security/src/main/java/io/airlift/security/client/ClientSecurityConfig.java

```diff
@@ -0,0 +1,92 @@
+package io.airlift.security.client;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+public class ClientSecurityConfig
+{
+    private String krb5ccPath;
+    private String krb5Keytab;
+    private String krb5ConfPath;
+    private String krb5ClientPrincipal;
+    private boolean authenticationEnabled;
+    private String serviceName;
+
+    public String getKrb5ccPath()
+    {
+        return krb5ccPath;
+    }
+
+    @Config("http.client.authentication.negotiate.krb5cc-path")
+    @ConfigDescription("Set kerberos credential cache path")
+    public ClientSecurityConfig setKrb5ccPath(String krb5ccPath)
+    {
+        this.krb5ccPath = krb5ccPath;
+        return this;
+    }
+
+    public String getKrb5Keytab()
+    {
+        return krb5Keytab;
+    }
+
+    @Config("http.client.authentication.negotiate.krb5-keytab")
+    @ConfigDescription("Set kerberos key table path")
+    public ClientSecurityConfig setKrb5Keytab(String krb5Keytab)
+    {
+        this.krb5Keytab = krb5Keytab;
+        return this;
+    }
+
+    public String getKrb5ClientPrincipal()
+    {
+        return krb5ClientPrincipal;
+    }
+
+    @Config("http.client.authentication.negotiate.krb5-client-principal")
```

Comment:
@martint, from Kerberos perspective, the client principal for a running service is '<client service name>/<host>'. The client service (in your context) needs to set this when using Presto client.

@zhenyuy-fb , how about using following names for consistency?
krb5-client-principal -> client-principal (config)
krb5-keytab -> keytab-path (config)
krb5ClientPrincipal -> clientPrincipal (code)
krb5Keytab -> keytabPath (code)